### ISSUE TITLE
CodeMeta support in Github

### ISSUE THREAD

**Issue by priya-gitTest** [codemeta/codemeta#359]
Hi,
Is there a plan to integrate CodeMeta in Github similar to  CITATION.cff File ?

P

**Comment by progval** [comment]
Hi,

Github has no such plans as far as I am aware, but you can contact their support to ask them.

**Comment by priya-gitTest** [comment]
Created a Ticket. Lets see what they reply. https://support.github.com/ticket/personal/0/2736548

**Comment by priya-gitTest** [comment]
Their COPilot suggested to look into this page, which looks like an interesting alternative to see how CodeMeta can be generated/aligned from this, if possible.. https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/about-community-profiles-for-public-repositories

**Comment by progval** [comment]
I don't see how this is related, "community profiles" are links to some text files while what you are suggesting (if I'm not mistaken) is for Github to generate APA/BibTeX/... from CodeMeta files in the repository.

**Comment by priya-gitTest** [comment]
Its not clear what data they use to generate it, and if that can converge with some fields of CodeMeta but for now, I have opened this discussion. If you think it needs correction, let me know . 

https://github.com/orgs/community/discussions/120212

**Comment by moranegg** [comment]
InvenioRDM & Zenodo are planning for this support to be available at the end of 2024 or start of 2025.
Applicable to any release including a codemeta.json file.

@arfon any plans for codemeta.json support in GitHub?

**Comment by arfon** [comment]
> @arfon any plans for codemeta.json support in GitHub?

Nothing immediate no, but there are (at least) two possible things we could be talking about here:

1. Teach GitHub to look for codemeta.json files and read them if discovered (e.g., similar to how [CITATION.cff ](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files)files are detected and parsed).
2. Get GitHub to generate codemeta.json files from the right-hand menu (that currently produces an APA or BibTeX formatted citation).

In both cases, we use the https://github.com/citation-file-format/ruby-cff library to read files. Depending upon which you're talking about, we would probably need to teach the `ruby-cff` library about codemeta files.

**Comment by moranegg** [comment]
@arfon Thanks.

So you would suggest using directly the `ruby-cff` library and not creating another library for `codemeta.json`?
If so, I can see if someone is game for doing a PR there.

**Comment by arfon** [comment]
Definitely. Even though the name (`ruby-cff` is a little awkward here), given we already have this as a 'paved path' for reading citation files in Ruby, I think it makes most sense (certainly for including in the GitHub codebase) for us to teach that library to understand these files.

/ cc @hainesr in case he feels very differently about this :-)
